tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1865 @@
+        self.execute_command('promiscuous enable')
+
+    def disable_promiscuous(self):
+        """Disable radio promiscuous operation."""
+        self.execute_command('promiscuous disable')
+
+    def get_txpower(self) -> int:
+        """Get the transmit power in dBm."""
+        line = self.__parse_str(self.execute_command('txpower'))
+        if not line.endswith(' dBm'):
+            raise UnexpectedCommandOutput([line])
+
+        return int(line.split()[0])
+
+    def set_txpower(self, val: int):
+        """Set the transmit power in dBm."""
+        self.execute_command(f'txpower {val}')
+
+    # TODO: fem
+    # TODO: fem lnagain
+    # TODO: fem lnagain <LNA gain>
+    # TODO: mac retries direct
+    # TODO: mac retries direct
+    # TODO: mac retries indirect
+    # TODO: mac retries indirect <number>
+
+    #
+    # Basic Node states and properties
+    #
+
+    def get_state(self) -> ThreadState:
+        """Get the current Thread state."""
+        return ThreadState(self.__parse_str(self.execute_command('state')))
+
+    def set_state(self, state: str):
+        """Try to switch to state detached, child, router or leader."""
+        self.execute_command(f'state {state}')
+
+    def get_rloc16(self) -> int:
+        """Get the Thread RLOC16 value."""
+        return self.__parse_int(self.execute_command('rloc16'), 16)
+
+    def get_router_id(self) -> int:
+        """Get the Thread Router ID value."""
+        return self.get_rloc16() >> 10
+
+    def prefer_router_id(self, routerid: int):
+        """Prefer a Router ID when solicit router id from Leader."""
+        self.execute_command(f'preferrouterid {routerid}')
+
+    def is_singleton(self) -> bool:
+        return self.__parse_values(self.execute_command('singleton'), true=True, false=False)
+
+    #
+    # RCP related utilities
+    #
+
+    def get_rcp_version(self):
+        return self.__parse_str(self.execute_command('rcp version'))
+
+    #
+    # Unsecure port utilities
+    #
+
+    def get_unsecure_ports(self) -> List[int]:
+        """all ports from the allowed unsecured port list."""
+        return self.__parse_int_list(self.execute_command('unsecureport get'))
+
+    def add_unsecure_port(self, port: int):
+        """Add a port to the allowed unsecured port list."""
+        self.execute_command(f'unsecureport add {port}')
+
+    def remove_unsecure_port(self, port: int):
+        """Remove a port from the allowed unsecured port list."""
+        self.execute_command(f'unsecureport remove {port}')
+
+    def clear_unsecure_ports(self):
+        """Remove all ports from the allowed unsecured port list."""
+        self.execute_command('unsecureport remove all')
+
+    #
+    # Leader configurations
+    #
+
+    def get_fixed_leader_partition_id(self) -> PartitionId:
+        """Get the fixed Thread Leader Partition ID."""
+        return PartitionId(self.__parse_int(self.execute_command('leaderpartitionid')))
```

Comment:
update to match changes in https://github.com/openthread/openthread/pull/5872

Reply:
We might need to check for API version and use the correct command to work with older openthread versions.

Reply:
fixed